pkg/reconciler/revision/resources/queue.go

```diff
@@ -270,3 +315,30 @@ func makeQueueContainer(rev *v1alpha1.Revision, loggingConfig *logging.Config, o
 		}},
 	}
 }
+
+func applyReadinessProbeDefaults(p *corev1.Probe, port int32) {
+	if p == nil {
+		return
+	}
+
+	if p.HTTPGet != nil {
+		p.HTTPGet.Host = localAddress
+		p.HTTPGet.Port = intstr.FromInt(int(port))
+
+		if p.HTTPGet.Scheme == "" {
+			p.HTTPGet.Scheme = corev1.URISchemeHTTP
+		}
+
+		p.HTTPGet.HTTPHeaders = append(p.HTTPGet.HTTPHeaders, corev1.HTTPHeader{
```

Comment:
Is this needed given that it is performed by the queue-proxy directly against the user-container?

Reply:
Yes.  If the user container is sensitive to this, we don't want to mess things up like Istio messed us up :(